src/org/mozilla/javascript/NativeJSON.java

```diff
@@ -294,10 +294,17 @@ private static Object str(Object key, Scriptable holder, StringifyState state) {
             value = getProperty(holder, ((Number) key).intValue());
         }
 
+        if (ScriptRuntime.isSymbol(value)) return Undefined.instance;
+
         if (value instanceof Scriptable && hasProperty((Scriptable) value, "toJSON")) {
             Object toJSON = getProperty((Scriptable) value, "toJSON");
             if (toJSON instanceof Callable) {
-                value = callMethod(state.cx, (Scriptable) value, "toJSON", new Object[] {key});
+                value =
+                        callMethod(
+                                state.cx,
+                                (Scriptable) value,
+                                "toJSON",
+                                new Object[] {ScriptRuntime.toString(key)});
```

Comment:
Perfromance-wise, would it make sense to do `key instanceof String ? key :  Integer.toString(key)` here, instead of  calling `ScriptRuntime.toString(key)`

Reply:
i think this is more or less what toString does already

Reply:
It does, just wondering if my example code would be slightly more efficient, as `key` can only be of type string or int (judging the code above)

Reply:
ahh, great, will update this